### ISSUE TITLE
No setup information found for ghc-8.4.3 when not installed ncurses version 5

### ISSUE THREAD

**Issue by ncaq** [commercialhaskell/stack#4144]
### General summary/comments (optional)

User environment do not have ncurses 5. 
But have ncurses 6.

So, `[CompilerBuild]` not include `CompilerBuildStandard`.
`[CompilerBuild]` is tinfo6 and ncurses6.

`stack setup` fails according to [stack-setup-2.yaml](https://raw.githubusercontent.com/fpco/stackage-content/master/stack/stack-setup-2.yaml).

### Steps to reproduce

For example:

1. Prepare an environment with ncurses version 6 instead of 5.(example Gentoo, Arch)
2. Run command `stack config set resolver lts`.
3. Make sure LTS is 12
4. Run command `stack blah`.

~~~yaml
packages: []
resolver: lts-12.0
~~~

### Expected

`stack setup` install ghc 8.4.3.

### Actual

~~~
% stack setup
Writing implicit global project config file to: /home/ncaq/.stack/global-project/stack.yaml
Note: You can change the snapshot via the resolver field there.
Using latest snapshot resolver: lts-12.0
Downloaded lts-12.0 build plan.
No setup information found for ghc-8.4.3 on your platform.
This probably means a GHC bindist has not yet been added for OS key 'linux64-ncurses6', 'linux64-tinfo6'.
Supported versions: ghc-7.8.4, ghc-7.10.2, ghc-7.10.3, ghc-8.0.1, ghc-8.0.2, ghc-8.2.1, ghc-8.2.2, ghc-8.4.1, ghc-8.4.2
~~~

### Stack version

```
% stack --version
Version 1.7.1, Git revision 681c800873816c022739ca7ed14755e85a579565 (5807 commits) x86_64 hpack-0.28.2
```

### Method of installation

* Official binary, downloaded from stackage.org or fpcomplete's package repository

### Workaround

User install ncurses version 5.

Example, case Gentoo, write `sys-libs/ncurses tinfo` in `package.use`, `sudo emerge ncurses:5`.

**Comment by ncaq** [comment]
This problem report on [Reddit](https://www.reddit.com/r/haskell/comments/8srw95/upcoming_stackage_lts_12_snapshot_with_ghc843/e17ce09/).
Arch Linux have problem too.

**Comment by ncaq** [comment]
It is the following place of the source code that judges whether it is the standard environment.
[stack/Setup.hs at 735240f5b99c343e5d86fa57db0893a3fd05f32a · commercialhaskell/stack](https://github.com/commercialhaskell/stack/blob/735240f5b99c343e5d86fa57db0893a3fd05f32a/src/Stack/Setup.hs#L608)

**Comment by borsboom** [comment]
Unfortunately there has not yet been a `fedora27` bindist released for ghc-8.4.3, which is needed to support distros the newer libtinfo6.  @bgamari, are there plans to build this like has been done for the previous ghc-8.4.* releases?

**Comment by wdanilo** [comment]
The same problem on Gentoo. Just fyi.

**Comment by bgamari** [comment]
Emanuel Borsboom <notifications@github.com> writes:

> Unfortunately there has not yet been a `fedora27` bindist released for
> ghc-8.4.3, which is needed to support distros the newer libtinfo6.
> @bgamari, are there plans to build this like has been done for the
> previous ghc-8.4.* releases?
>
Yes, I can try to get this done tonight.

**Comment by Kasama** [comment]
any news on this issue? Still experiencing it on Arch Linux

**Comment by bgamari** [comment]
Emanuel Borsboom <notifications@github.com> writes:

> Unfortunately there has not yet been a `fedora27` bindist released for
> ghc-8.4.3, which is needed to support distros the newer libtinfo6.
> @bgamari, are there plans to build this like has been done for the
> previous ghc-8.4.* releases?
>
I had troubles with my Fedora 27 build environment and there were no
complaints regarding its absence, so I assumed that there was no demand.

Today I took some time to rebuild my build environment and build 8.4.3
bindists. They are now available at the usual place.

Sorry for the delay!

**Comment by borsboom** [comment]
Thanks!  I think once Stackage LTS 12 was released, which uses GHC 8.4.3, people really noticed that this bindist was missing.

I've added the metadata for this build to Stack's setup metadata, so Stack will now be able to find it on systems with libtinfo6/libncurses6.

@ncaq Please confirm that this works, and close the issue if so.

**Comment by ncaq** [comment]
Great, I confirmed, I close it.
